### ISSUE TITLE
Implement some sort of built-in check to avoid errors on frequent asynchronous data updates

### ISSUE THREAD

**Issue by ptorrent** [amcharts/amcharts4#1185]
Hello,

It's not possible to call addData just after chart.data is set.

https://codepen.io/ptorrent/pen/bQWNwr?editors=1010

```javascript

   chart.data = data
   chart.addData({value : Math.random()*100, date : ++I},1)
  
```

**Comment by martynasma** [comment]
Why would you do that?

**Comment by ptorrent** [comment]
I can do it by other way....

It was working before I upgrade the library....

**Comment by martynasma** [comment]
Well, if it was working before, doesn't mean its right :)

To be on the safe side, wait for `datavalidated` event or check for `dataInvalid` before adding new data.

```
if (!chart.dataInvalid) {
  // addData
}
```

Or

```
if (chart.dataInvalid) {
  chart.events.once("datavalidated", function() {
    // addData
    // ...
  });
}
```

**Comment by ptorrent** [comment]
Thanks for your help,

Is there a reason why you don't do it from your side ?

**Comment by martynasma** [comment]
Fair enough.

I'm redoing this thread as an enhancement. I guess we can put some checks in place.

**Comment by martynasma** [comment]
Fixed in [4.3.15](https://github.com/amcharts/amcharts4/releases/tag/4.3.15).

## [4.3.15] - 2019-04-06

### Added
- `startendchanged` event added to `Component`. Fired when `start` or `end` is changed. Unlike `startchanged`/`endchanged` this event is fired not immediately but at the end of a cycle.

### Changed
- `Sunburst` chart's defaalt `radius` set to `95%`.

### Fixed
- `TreeMap` could produce a JS error if data was changed during interaction with the map.
- If `Component.addData()` called before chart was initialized, error happened.
- 3D columns with reversed base axis had wrong z-indexes.
- `DateAxis`-based chart with single data item got into infinite loop if `startLocation` and `endLocation` were set to `0.5` or `strictMinMax` was set to `true`.
- Tooltips of series could overlap if series used different `DateAxis` axes.
- Chart with multiple `DateAxis` and no animated theme was not properly zooming with scrollbar.
- Implemented a workaround for MSIE which was not updating series masks on chart resize.
- Disposing `XYCursor` will now remove series/axes tooltips that might be currently displayed because of it.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
